Doctrine/EmailUpdateListener.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace FOS\UserBundle\Doctrine;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\Common\Persistence\Event\LifecycleEventArgs;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use FOS\UserBundle\Model\UserInterface;
+
+
+/**
+ * Class EmailUpdateListener
+ * @package FOS\UserBundle\Doctrine
+ */
+class EmailUpdateListener implements EventSubscriber
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * Constructor
+     *
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    public function getSubscribedEvents()
+    {
+        return array(
+            'preUpdate',
+        );
+    }
+
+    /**
+     * Pre update listener based on doctrine common
+     *
+     * @param LifecycleEventArgs $args
+     */
+    public function preUpdate(LifecycleEventArgs $args)
+    {
+        $object = $args->getObject();
+        if ($object instanceof UserInterface) {
+
```

Comment:
"cast" the $object here.
```
/** @var UserInterface $user */
$user = $object;
```

and then use $user instead of $object further below.